### PR TITLE
ORT 1.11 mobile doc updates

### PR DESCRIPTION
**Pull request by edgchen1**
**Description**
Various mobile-related documentation updates for 1.11.

https://edgchen1.github.io/onnxruntime/

**Motivation and Context**
Update documentation for 1.11.

### CODE REVIEW

File: docs/execution-providers/NNAPI-ExecutionProvider.md [hunk 1]
```diff
@@ -22,7 +22,9 @@ Accelerate ONNX models on Android devices with ONNX Runtime and the NNAPI execut
 The NNAPI Execution Provider (EP) requires Android devices with Android 8.1 or higher. It is recommended to use Android devices with Android 9 or higher to achieve optimal performance.
 
 ## Install
-Pre-built packages of ONNX Runtime Mobile with NNAPI EP for Android are published on Maven: [com.microsoft.onnxruntime:onnxruntime-mobile](https://mvnrepository.com/artifact/com.microsoft.onnxruntime/onnxruntime-mobile).
+Pre-built packages of ONNX Runtime with NNAPI EP for Android are published on Maven.
+
+See [here](../install/index.md#install-on-android) for installation instructions.
```
Comment (skottmckay) [inline docs/execution-providers/NNAPI-ExecutionProvider.md:28]: We're removing the link to Maven but I didn't see any other place where the user is provided that info. Can we add 2 bullet points - one with the link to maven for the mobile package and one with the link to the full package?

File: docs/install/index.md [hunk 2]
```diff
@@ -72,9 +72,13 @@ dotnet add package Microsoft.AI.MachineLearning
...
 
-The installation instructions in this section use pre-built packages that include support for selected operators and ONNX opset versions based on the requirements of popular models. Your model must only use the [opsets and operators](../reference/operators/mobile_package_op_type_support_1.9.md) supported by the pre-built package.
+Unless stated otherwise, the installation instructions in this section refer to pre-built packages that include support for selected operators and ONNX opset versions based on the requirements of popular models. These packages may be referred to as "mobile packages". If you use mobile packages, your model must only use the supported [opsets and operators](../reference/operators/mobile_package_op_type_support_1.9.md).
 
-If the pre-built package is too large, or does not include the operators in your model/s, you can create a [custom build](../build/custom.md).
+Another type of pre-built package has full support for all ONNX opsets and operators, at the cost of larger binary size. These packages are referred to as "full packages".
+
+If the pre-built mobile package is too large, you can create a [custom build](../build/custom.md).
+
```
Comment (skottmckay) [inline docs/install/index.md:80]: nit: need to say why creating a custom build would help

File: docs/install/index.md [hunk 3]
```diff
@@ -72,9 +72,13 @@ dotnet add package Microsoft.AI.MachineLearning
...
+If the pre-built mobile package does not include the operators in your model/s, you can either use the full package if available, or create a custom build.
```
Comment (skottmckay) [inline docs/install/index.md:82]: Could also be the model opset is the issue. Can add link when the new doco on the onnx opset update helper is checked in.

File: docs/install/index.md [hunk 4]
```diff
@@ -105,22 +109,26 @@ npm install onnxruntime-react-native
 
 ### Install on iOS
 
-In your CocoaPods `Podfile`, add the `onnxruntime-mobile-c` or `onnxruntime-mobile-objc` pod depending on which API you wish to use.
+In your CocoaPods `Podfile`, add the `onnxruntime-c`, `onnxruntime-mobile-c`, `onnxruntime-objc`, or `onnxruntime-mobile-objc` pod, depending on whether you want to use a full or mobile package and which API you want to use.
```
Comment (skottmckay) [inline docs/install/index.md:113]: nit: maybe split this up a bit more clearly. 

e.g. 
In your CocoaPods `Podfile`, add the cocoapods for the onnxruntime package and API you wish to use
  - full: `onnxruntime-c`, and optionally `onnxruntime-objc`
  - mobile: `onnxruntime-mobile-c`, and optionally `onnxruntime-mobile-objc`

**Comment by skottmckay** [comment]
> parent: Performance

This whole page can probably go away as runtime optimization makes it pretty redundant.

---
Refers to: docs/performance/mobile-performance-tuning.md:3 in 487593a. [](commit_id = 487593acb4601ad21a9836675e1e560906a0cefd, deletion_comment = False)

File: docs/reference/ort-format-models.md [hunk 5]
```diff
@@ -16,9 +16,31 @@ redirect_from: /docs/tutorials/mobile/model-conversion, /docs/tutorials/mobile/m
...
+Consequently, the graph optimizers do not run the same way at runtime for ORT format models as they do for ONNX format models.
+
+To allow for dynamic graph optimization at runtime, some graph optimizers support additional modes to save and load saved runtime optimizations to and from the ORT format model.
+When converting from ONNX to ORT format, the potential optimizations are identified (1) and their effects are saved alongside the graph (without those optimizations applied) in the ORT format model.
+Later, when loading the ORT format model with saved runtime optimizations, the effects of potential optimizations are applied (2) if the potential optimizations are still applicable.
+
+In a reduced size build (an [extended minimal build](../build/custom.md#minimal-build)), only enough implementation to support (2) is included, reducing the graph optimizers' binary size.
+
+Note that saving runtime optimizations is optional. Alternatively, it is possible to save a fully optimized ORT format model, but the graph's optimizations would be more hardware-dependent and less flexible at runtime.
```
Comment (skottmckay) [inline docs/reference/ort-format-models.md:43]: >Note that saving runtime optimizations is optional. Alternatively, it is possible to save a fully optimized ORT format model, but the graph's optimizations would be more  [](http://example.com/codeflow?start=0&length=170)

nit: `would be more` -> `may be`

File: docs/tutorials/mobile/index.md [hunk 6]
```diff
@@ -18,21 +18,47 @@ ONNX Runtime gives you a variety of options to add machine learning to your mobi
...
+   |-|-|
+   |onnxruntime-mobile|3.3 MB|
+   |onnxruntime-android|12 MB|
+
+   ONNX Runtime 1.11.0 iOS package `onnxruntime.xcframework/ios-arm64/onnxruntime.framework/onnxruntime` file size:
+   |Package|Size|
+   |-|-|
+   |onnxruntime-mobile-c|22 MB|
+   |onnxruntime-c|48 MB|
+
```
Comment (skottmckay) [inline docs/tutorials/mobile/index.md:52]: This possibly needs to be qualified - the iOS package is a static library that will have a reduced binary size impact when compiled into your app.

File: docs/tutorials/mobile/index.md [hunk 7]
```diff
@@ -18,21 +18,47 @@ ONNX Runtime gives you a variety of options to add machine learning to your mobi
...
+   If the binary size of the full package is acceptable, using the full package is recommended because it is easier to use.
+   Otherwise, consider using the mobile package or a custom build.
+
+   To give an idea of the binary size difference between mobile and full packages:
+
+   ONNX Runtime 1.11.0 Android package `jni/arm64-v8a/libonnxruntime.so` file size:
+   |Package|Size|
+   |-|-|
+   |onnxruntime-mobile|3.3 MB|
+   |onnxruntime-android|12 MB|
```
Comment (skottmckay) [inline docs/tutorials/mobile/index.md:45]: That's a larger size than expected. Wasn't the mobile package around 2.5MB a couple of releases ago? 

We did add some more opsets and runtime optimization I guess, but I wasn't expecting it to be > 3MB.

**Comment by skottmckay** [comment]
> 

nit: the content here doesn't clearly answer the question 'how do I optimize my application'. 

We could say the ORT format model uses less memory (no schemas required and limited optimizers to run). 

Using the mobile package or doing a custom build will reduce binary size.

Model size is pretty equivalent, so not sure that should be mentioned under either binary size or network usage.

---
Refers to: docs/tutorials/mobile/index.md:76 in 487593a. [](commit_id = 487593acb4601ad21a9836675e1e560906a0cefd, deletion_comment = False)

**Comment by edgchen1** [comment]
> 

yes, that question is kind of broad. maybe it needs to be more specific.

---
In reply to: [1085321129](https://github.com/microsoft/onnxruntime/pull/11062#issuecomment-1085321129) [](http://example.com/codeflow?ancestors=1085321129,1085321129)

---
Refers to: docs/tutorials/mobile/index.md:76 in 487593a. [](commit_id = 487593acb4601ad21a9836675e1e560906a0cefd, deletion_comment = False)

File: docs/tutorials/mobile/index.md (hunk 7 above)
Comment (edgchen1) [reply docs/tutorials/mobile/index.md:45]: checked the packages here:
[https://search.maven.org/artifact/com.microsoft.onnxruntime/onnxruntime-mobile](https://search.maven.org/artifact/com.microsoft.onnxruntime/onnxruntime-mobile)

1.10.0: 2.9 MB
1.9.0: 2.9 MB
1.8.2: 2.7 MB

**Comment by edgchen1** [comment]
> parent: Performance

It still applies to older versions though. We could have entirely new instructions for 1.11+ that refer to the mobile helpers. Not sure how to best organize this though. May update this in another PR.

---
In reply to: [1085302811](https://github.com/microsoft/onnxruntime/pull/11062#issuecomment-1085302811) [](http://example.com/codeflow?ancestors=1085302811,1085302811)

---
Refers to: docs/performance/mobile-performance-tuning.md:3 in 487593a. [](commit_id = 487593acb4601ad21a9836675e1e560906a0cefd, deletion_comment = False)

File: docs/install/index.md [hunk 8]
```diff
@@ -72,9 +72,13 @@ dotnet add package Microsoft.AI.MachineLearning
...
+Unless stated otherwise, the installation instructions in this section refer to pre-built packages that include support for selected operators and ONNX opset versions based on the requirements of popular models. These packages may be referred to as "mobile packages". If you use mobile packages, your model must only use the supported [opsets and operators](../reference/operators/mobile_package_op_type_support_1.9.md).
 
-If the pre-built package is too large, or does not include the operators in your model/s, you can create a [custom build](../build/custom.md).
+Another type of pre-built package has full support for all ONNX opsets and operators, at the cost of larger binary size. These packages are referred to as "full packages".
+
+If the pre-built mobile package is too large, you can create a [custom build](../build/custom.md). A custom build can contain just the opsets and operators that you need, so it may be smaller.
+
```
Comment (skottmckay) [inline docs/install/index.md:80]: nit: It should definitely be smaller so 'may be smaller' doesn't quite convey that it's worth spending the time on the custom build.

For it to be larger, you would need to use more operators/opsets/types than the pre-built package - which has a config based on ~60 models and supports 4 different onnx opsets.

File: docs/tutorials/mobile/index.md (hunk 7 above)
Comment (skottmckay) [reply docs/tutorials/mobile/index.md:45]: I expected about 150kb from runtime optimization and QDQ support so a 400kb increase is surprising. There are some additional MLAS kernels but usually those are small. Definitely something to look into.

**Review by skottmckay — APPROVED** [review]
:shipit:

**Review by raghavthind2005 — APPROVED** [review]

**Pull request merged**